src/runtime/security/negotiation_manager.cpp

```diff
@@ -102,14 +102,25 @@ void negotiation_manager::on_rpc_disconnected(rpc_session *session)
     }
 }
 
+// `on_rpc_recv_msg` and `on_rpc_recv_msg` will be called by both server and client session
+// For server sessioin, it can receive message if mandatory_auth is false
+// Because mandatory_auth is a server-side config only, so it doesn't have the same effect for
+// client session
```

Comment:
```suggestion
// `on_rpc_send_msg` and `on_rpc_recv_msg` will be called by both server and client session.
// For server session, it can bypass negotiation if mandatory_auth is false.
// mandatory_auth is a server-side config only, it doesn't have the same effect for
// client session. 
```